### DISCUSSION TITLE
Clarification is needed for the chat.bsky.moderation.GetMessageContext endpoint

### DISCUSSION THREAD

**Discussion by MadhuGowthaman** [bluesky-social/atproto#3185] [Bluesky Lexicons]
Hello, everyone.

I attempted to retrieve single message data using the "chat.bsky.moderation.getMessageContext" endpoint, but received the result as {"error":"UntrustedIss","message":"Untrusted issuer"}.

Could someone please help me?

Thank you.

**Comment by lithp** [comment]
I know very little about atproto, but this sounds like an issue with the jwt you're passing in. Here's the spec on how to authenticate with services: https://atproto.com/specs/xrpc#authentication

It looks like the chat service is not open source so unfortunately I can't tell you much more than that

From perusing through the atproto code, which I assume that chat service is probably based on... this error might occur when you've built your JWT with the wrong iss suffix:
- [here](https://github.com/bluesky-social/atproto/blob/main/packages/bsky/src/auth-verifier.ts#L303)

The spec mentions the iss field sometimes needs to be in the form "[did]#atproto_chat", or something like that

This seems entirely undocumented. It might be easiest to just run a PDS, run its traffic through a mitmproxy, and look directly at the requests it's making.
